src/Inkluzitron/Modules/BdsmTestOrg/BdsmModule.cs

```diff
@@ -253,83 +224,64 @@ private async Task SearchAndTextAnswerAsync(bool showAllMode, params string[] qu
+            if (!testResultMatch.Success)
             {
                 await ReplyAsync(Settings.InvalidFormatMessage);
                 return;
             }
 
-            var testResultMatch = testResultMatches.Single();
-            var testResultItems = testResultMatch.Groups["results"].Value;
-            var testResultLink = testResultMatch.Groups["link"].Value;
+            link = testResultMatch.Value;
 
-            if (await DbContext.BdsmTestOrgQuizResults.AsQueryable().AnyAsync(r => r.Link == testResultLink))
+            if (await DbContext.BdsmTestOrgQuizResults.AsQueryable().AnyAsync(r => r.Link == link))
             {
                 await ReplyAsync(Settings.LinkAlreadyPresentMessage);
                 return;
             }
 
-            var itemsMatch = TestResultItemRegex.Matches(testResultItems);
+            var testid = testResultMatch.Groups[1].Value;
 
-            var testResult = new BdsmTestOrgQuizResult
+            var requestData = new Dictionary<string, string>
             {
-                SubmittedAt = DateTime.UtcNow,
+                { "uauth[uid]", Settings.ApiKey.Uid },
+                { "uauth[salt]", Settings.ApiKey.Salt },
+                { "uauth[authsig]", Settings.ApiKey.AuthSig },
+                { "rauth[rid]", testid }
+            };
+
+            var content = new FormUrlEncodedContent(requestData);
+
+            var response = await HttpClient.PostAsync("https://bdsmtest.org/ajax/getresult", content);
+
+            var responseData = await response.Content.ReadAsStringAsync();
+            var testResult = JsonConvert.DeserializeObject<Result>(responseData);
+
+            var testResultDb = new BdsmTestOrgQuizResult
+            {
+                SubmittedAt = testResult.Date,
                 SubmittedByName = Context.Message.Author.Username,
                 SubmittedById = Context.Message.Author.Id,
-                Link = testResultLink
+                Link = link
             };
 
-            foreach (Match match in itemsMatch)
+            foreach (var trait in testResult.Traits)
             {
-                var traitName = match.Groups["trait"].Value;
-                if (!Settings.TraitList.Contains(traitName))
-                {
-                    await ReplyAsync(Settings.InvalidTraitMessage);
-                    return;
-                }
-
-                if (!TryParseTraitPercentage(match.Groups["pctg"].Value, out var traitPercentage))
-                {
-                    await ReplyAsync(Settings.InvalidPercentageMessage);
-                    return;
-                }
+                if (!Settings.Traits.Any(t => t.Id == trait.Id)) continue;
 
-                testResult.Items.Add(new QuizDoubleItem
+                testResultDb.Items.Add(new QuizDoubleItem
                 {
-                    Key = traitName,
-                    Value = traitPercentage
+                    Key = trait.Name,
+                    Value = trait.Score / 100.0
```

Comment:
Ocenil bych tu nějakou validaci, aby tam nemohlo vlézt něco mimo rozsah.

Reply:
Přidal jsem validaci v 4bdf308. V jednoduché formě oříznutí intervalů.

Napiš, jestli je to takhle ok.